planner/core/plan_stats.go

```diff
@@ -33,9 +33,10 @@ import (
 
 type collectPredicateColumnsPoint struct{}
 
-func (collectPredicateColumnsPoint) optimize(_ context.Context, plan LogicalPlan, _ *logicalOptimizeOp) (LogicalPlan, error) {
+func (collectPredicateColumnsPoint) optimize(_ context.Context, plan LogicalPlan, _ *logicalOptimizeOp) (LogicalPlan, bool, error) {
```

Comment:
I think we need to add a TODO here. The sync load/col usage analysis should be moved out of from the rule iteration.

Reply:
It relies on the predicate push-down currently. But i think it will be better if we detach it and move it out as a pre-opt operation.

Reply:
There is two stats rule in Rule List : collectPredicateColumnsPoint,  syncWaitStatsLoadPoint.
My origin plan about New Rule Engine is separate those stats rule from Rule List.

We can come down and talk about the risks. Because it seems that some rules rely on statistical information?